pkg/metrics/query.go

```diff
@@ -0,0 +1,218 @@
+package metrics
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	kapierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/prometheus/common/model"
+)
+
+const (
+	serviceCAinjectionDataKey             = "service-ca.crt"
+	serviceCAinjectCABundleAnnotationName = "service.beta.openshift.io/inject-cabundle"
+	metricsBaseURL                        = "https://prometheus-k8s.openshift-monitoring.svc:9091"
+)
+
+type PromQuery struct {
+	Client        *kubernetes.Clientset
+	Query         string
+	ConfigMapName string
+	NsName        string
+	URL           string
+	BearerToken   string
+}
+
+func NewPromQuery(client *kubernetes.Clientset, query, cmName, nsName string) (*PromQuery, error) {
```

Comment:
I suggest separating ca bundle retrieval from prometheus querying. Ideally something like the following:

```golang
// RetrieveServiceCABundle would internally create an annotated configmap in the provided namespace, retrieve the configmap from it, and delete it. 
caBundle, err := RetrieveServiceCABundle(client, namespace)

// NewPrometheusClient would do the work of initializing a client capable of making queries
prometheusClient, err := NewPrometheusClient(client, namespace, cabundle)

// Actually run the query
result, err := prometheusClient.Query(...)
```

Reply:
updated, _much better thank you_

Reply:
no longer required, using InsecureSkipVerify, no CAbundle retrieval